cli/testflinger_cli/errors.py

```diff
@@ -17,3 +17,15 @@ def __init__(self, path: PathLike[str]) -> None:
         :param path: The path of the file that is in a snap-private directory.
         """
         super().__init__(f"File {path} is in a snap-private directory.")
+
+
+class TFStatusError(Exception):
+    """Exception thrown when unable to retrieve status
+    information from TF Server.
+    """
+
+    def __init__(self, endpoint: str) -> None:
+        super().__init__(
+            f"Unable to retrieve {endpoint} status from the server, check "
+            "your connection or try again later."
+        )
```

Comment:
Love this, can we rename it to `UnknownStatusError`, that name is more descriptive of what the exception actually is.

I also recommend rephrasing the docstring to make it fit within the first line.

```suggestion
class UnknownStatusError(Exception):
    """Exception thrown when unable to retrieve a status from the server."""

    def __init__(self, endpoint: str) -> None:
        super().__init__(
            f"Unable to retrieve {endpoint} status from the server, check "
            "your connection or try again later."
        )
```